--- /usr/share/backupninja/rsync        2012-06-29 20:43:57.000000000 +0000
+++ rsync       2014-11-05 08:10:41.815388627 +0000
@@ -1138,7 +1138,7 @@
   info "Syncing $SECTION on $dest_path..."
   debug $nice $rsync ${rsync_options[@]} $filelist_flag $excludes $batch_option $orig $dest_path
   set_pipefail
-  $nice su -c "$rsync ${rsync_options[@]} --delete-excluded $filelist_flag $excludes $batch_option $orig $dest_path" | tee -a $log
+  $nice su -c "`echo $rsync ${rsync_options[@]} --delete-excluded $filelist_flag $excludes $batch_option $orig $dest_path`" | tee -a $log

   check_rsync_exit_status $?
   restore_pipefail
