--- /usr/share/backupninja/pgsql        2014-11-19 08:20:28.036649513 +0000
+++ pgsql.new   2014-11-19 08:08:40.796620944 +0000
@@ -130,9 +130,11 @@
       fi
    else
       if [ "$compress" == "yes" ]; then
-         execstr="$PGSQLDUMPALL | $GZIP $GZIP_OPTS > '$backupdir/${localhost}-all.sql.gz'"
+         backupfile=$backupdir/${localhost}-all.sql.gz
+         execstr="$PGSQLDUMPALL | $GZIP $GZIP_OPTS > '${backupfile}'"
       else
-         execstr="$PGSQLDUMPALL > '$backupdir/${localhost}-all.sql'"
+         backupfile=$backupdir/${localhost}-all.sql
+         execstr="$PGSQLDUMPALL > '${backupfile}'"
       fi
    fi
    debug "$execstr"
@@ -193,9 +195,11 @@
       dumpcmd=""
       globalscmd=""
       if [ "$compress" == "yes" ]; then
-         dumpcmd="set -o pipefail ; $PGSQLDUMP --format=$format ${disablecustomcompress} $db | $GZIP $GZIP_OPTS > '$backupdir/${db}.${dumpext}.gz'"
+         backupfile = $backupdir/${db}.${dumpext}.gz
+         dumpcmd="set -o pipefail ; $PGSQLDUMP --format=$format ${disablecustomcompress} $db | $GZIP $GZIP_OPTS > '$backupfile'"
       else
-         dumpcmd="$PGSQLDUMP --format=$format ${disablecustomcompress} $db > '$backupdir/${db}.${dumpext}'"
+         backupfile = $backupdir/${db}.${dumpext}
+         dumpcmd="$PGSQLDUMP --format=$format ${disablecustomcompress} $db > '$backupfile'"
       fi
       if [ $usevserver = yes ]; then
          execstr="$VSERVER $vsname exec su - $PGSQLUSER -s /bin/bash -c \"$dumpcmd\""
@@ -217,5 +221,11 @@
    done
 fi

+## Rotate old backups
+if [ -e "$backupfile" ]; then
+  mv ${backupfile} ${backupdir}/${PGHOST}-$(date +%Y%m%d-%H%M%S).sql.gz
+  find ${backupdir} -name ${PGHOST}-* |sort -r | tail -n +8 |grep ${PGHOST} |xargs rm -r
+fi
+
 return 0

