--- /usr/share/backupninja/pgsql        2023-02-12 20:01:53.000000000 +0000
+++ pgsql       2024-12-03 13:42:20.179081964 +0000
@@ -9,6 +9,14 @@
 getconf compress yes
 # format maps to pg_dump --format= option, old/default was plain
 getconf format plain
+getconf PGHOST
+getconf PGUSER
+getconf PGPASSWORD
+export PGSQLUSER=root
+export PGHOST
+export PGUSER
+export PGPASSWORD
+echo "Using PGHOST=$PGHOST, PGUSER=$PGUSER"

 localhost=`hostname`

@@ -64,9 +72,9 @@
 # if $databases = all, use pg_dumpall
 if [ "$databases" == "all" ]; then
    if [ "$compress" == "yes" ]; then
-      execstr="su - $PGSQLUSER -s /bin/bash -c \"set -o pipefail ; $PGSQLDUMPALL | $GZIP $GZIP_OPTS > '$backupdir/${localhost}-all.sql.gz'\""
+      execstr="$PGSQLDUMPALL | $GZIP $GZIP_OPTS > '$backupdir/${localhost}-all.sql.gz'"
    else
-      execstr="su - $PGSQLUSER -c \"$PGSQLDUMPALL > '$backupdir/${localhost}-all.sql'\""
+      execstr="$PGSQLDUMPALL > '$backupdir/${localhost}-all.sql'"
    fi
    debug "$execstr"
    if [ ! $test ]; then
